scipy/optimize/_highs/cython/src/_highs_wrapper.pyx

```diff
@@ -1,736 +0,0 @@
-        lp.a_matrix_.index_.assign(aindex_ptr, aindex_ptr + numnz)
-        lp.a_matrix_.value_.assign(avalue_ptr, avalue_ptr + numnz)
-    else:
-        lp.a_matrix_.start_.empty()
-        lp.a_matrix_.index_.empty()
-        lp.a_matrix_.value_.empty()
-
-    # Create the options
-    cdef Highs highs
-    apply_options(options, highs)
-
-    # Make a Highs object and pass it everything
-    cdef HighsModelStatus err_model_status = HighsModelStatusNOTSET
-    cdef HighsStatus init_status = highs.passModel(lp)
-    if init_status != HighsStatusOK:
-        if init_status != HighsStatusWarning:
-            err_model_status = HighsModelStatusMODEL_ERROR
-            return {
-                'status': <int> err_model_status,
-                'message': highs.modelStatusToString(err_model_status).decode(),
-            }
-
-    # Solve the LP
-    cdef HighsStatus run_status = highs.run()
-    if run_status == HighsStatusError:
-        return {
-            'status': <int> highs.getModelStatus(),
-            'message': highsStatusToString(run_status).decode(),
-        }
-
-    # Extract what we need from the solution
-    cdef HighsModelStatus model_status = highs.getModelStatus()
-
-    # We might need an info object if we can look up the solution and a place to put solution
-    cdef HighsInfo info = highs.getHighsInfo() # it should always be safe to get the info object
-    cdef HighsSolution solution
-    cdef HighsBasis basis
-    cdef double[:, ::1] marg_bnds = np.zeros((2, numcol))  # marg_bnds[0, :]: lower
-
-    # Failure modes:
-    #     LP: if we have anything other than an Optimal status, it
-    #         is unsafe (and unhelpful) to read any results
-    #    MIP: has a non-Optimal status or has timed out/reached max iterations
-    #             1) If not Optimal/TimedOut/MaxIter status, there is no solution
-    #             2) If TimedOut/MaxIter status, there may be a feasible solution.
-    #                if the objective function value is not Infinity, then the
-    #                current solution is feasible and can be returned.  Else, there
-    #                is no solution.
-    mipFailCondition = model_status not in {
-        HighsModelStatusOPTIMAL,
-        HighsModelStatusREACHED_TIME_LIMIT,
-        HighsModelStatusREACHED_ITERATION_LIMIT,
-    } or (model_status in {
-        HighsModelStatusREACHED_TIME_LIMIT,
-        HighsModelStatusREACHED_ITERATION_LIMIT,
-    } and (info.objective_function_value == HIGHS_CONST_INF))
-    lpFailCondition = model_status != HighsModelStatusOPTIMAL
-    if (highs.getLp().isMip() and mipFailCondition) or (not highs.getLp().isMip() and lpFailCondition):
-        return {
-            'status': <int> model_status,
-            'message': f'model_status is {highs.modelStatusToString(model_status).decode()}; '
-                       f'primal_status is {utilBasisStatusToString(<HighsBasisStatus> info.primal_solution_status).decode()}',
```

Comment:
This seems to have been a bug to begin with, for example (using the documentation):
```python
import numpy as np
from scipy.optimize import linprog

# Objective function vector
c = np.array([-29.0, -45.0, 0.0, 0.0])

# Inequality constraints matrix
A_ub = np.array([
    [1.0, -1.0, -3.0, 0.0],
    [-2.0, 3.0, 7.0, -3.0]
])
# Inequality constraints vector
b_ub = np.array([5.0, -10.0])

# Equality constraints matrix
A_eq = np.array([
    [2.0, 8.0, 1.0, 0.0],
    [4.0, 4.0, 0.0, 1.0]
])
# Equality constraints vector
b_eq = np.array([60.0, 60.0])

# Bounds for each variable
x0_bounds = (0, None)
x1_bounds = (0, 5.0)
x2_bounds = (-np.inf, 0.5)  # +/- np.inf can be used instead of None
x3_bounds = (-3.0, None)
bounds = [x0_bounds, x1_bounds, x2_bounds, x3_bounds]

# Linear programming solution
result = linprog(c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq, bounds=bounds)

# Display message from the optimization result
print(result.message)
```
Which would previously report:
`The problem is infeasible. (HiGHS Status 8: model_status is Infeasible; primal_status is At lower/fixed bound)`

However, this is actually not true, and forcing the primal_solution_status instead of using `highs.solutionStatusToString(info.primal_solution_status)` shouldn't be happening. The following (by inserting a breakpoint):

```python
(Pdb) basis.row_status
[<HighsBasisStatus.kUpper: 2>, <HighsBasisStatus.kUpper: 2>, <HighsBasisStatus.kLower: 0>, <HighsBasisStatus.kLower: 0>]
(Pdb) basis.col_status
[<HighsBasisStatus.kBasic: 1>, <HighsBasisStatus.kBasic: 1>, <HighsBasisStatus.kBasic: 1>, <HighsBasisStatus.kLower: 0>]
(Pdb) [_h.Highs_().basisStatusToString(x) for x in basis.col_status]
['Basic', 'Basic', 'Basic', 'At lower/fixed bound']
(Pdb) [_h.Highs_().basisStatusToString(x) for x in basis.row_status]
['At upper bound', 'At upper bound', 'At lower/fixed bound', 'At lower/fixed bound']
```

Perhaps @mdhaber or @mckib2 know more about why the choice was made to cast `primal_solution_status` (a `SolutionStatus` object) to a `HighsBasisStatus` object?

Note that, `SolutionStatus` objects are defined by:

```cpp
  py::enum_<SolutionStatus>(m, "SolutionStatus")
      .value("kSolutionStatusNone", SolutionStatus::kSolutionStatusNone)
      .value("kSolutionStatusInfeasible",
             SolutionStatus::kSolutionStatusInfeasible)
      .value("kSolutionStatusFeasible",
             SolutionStatus::kSolutionStatusFeasible);
```

While a `HighsBasisStatus` object contains:

```cpp
  py::enum_<HighsBasisStatus>(m, "HighsBasisStatus")
      .value("kLower", HighsBasisStatus::kLower)
      .value("kBasic", HighsBasisStatus::kBasic)
      .value("kUpper", HighsBasisStatus::kUpper)
      .value("kZero", HighsBasisStatus::kZero)
      .value("kNonbasic", HighsBasisStatus::kNonbasic);
```

Reply:
I agree, this looks like a bug as written.  There shouldn't be any reason to cast to a `HighsBasisStatus` type.

Reply:
Awesome, thanks for verifying, resolving as fixed :)